lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,390 @@
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.RulesTable = cryptoRandomString({ length: 10 });
+
+  const workflowfile = `${process.env.stackName}/workflows/${workflow}.json`;
+  const messageTemplateKey = `${process.env.stackName}/workflow_template.json`;
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  rulesModel = new Rule();
+  await rulesModel.createTable();
+  await createBucket(process.env.system_bucket);
+
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: new Date(2020, 11, 17),
+  });
+
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
+
+  await Promise.all([
+    putJsonS3Object(
+      process.env.system_bucket,
+      messageTemplateKey,
+      { meta: 'meta' }
+    ),
+    putJsonS3Object(
+      process.env.system_bucket,
+      workflowfile,
+      { testworkflow: 'workflow-config' }
+    ),
+  ]);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex('rules').del();
+  await t.context.knex('providers').del();
+  await t.context.knex('collections').del();
+});
+
+test.after.always(async (t) => {
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await rulesModel.deleteTable();
+
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('getCumulusId throws ColumnDoesNotExist if table does not have a cumulus_id column', async (t) => {
+  const { knex } = t.context;
+  const result = getCumulusId(
+    {
+      name: fakeCollection.name,
+      version: fakeCollection.version,
+    },
+    'sometable',
+    knex
+  );
+  await t.throwsAsync(result, { name: 'ColumnDoesNotExist' });
+});
+
+test.serial('migrateRuleRecord correctly migrates rule record', async (t) => {
+  const { knex } = t.context;
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  await migrateFakeCollectionRecord(fakeCollection, knex);
+  await migrateFakeProviderRecord(fakeProvider, knex);
```

Comment:
I'd like to remove the use of these migration functions for collections/providers in favor of creating the records directly via Knex. Otherwise, this test is now inadvertently also testing the migration logic for collections/providers, so it's not isolated from bugs in that logic.

Reply:
Yes, I agree with you. I've written the functions `migrateFakeCollectionRecord` and `migrateFakeProviderRecord` which will call knex directly https://github.com/nasa/cumulus/pull/2000/files#diff-d7b0058e0252e8083289a57a228dc6cec19ad3877e20ff8c22626a419ae27affR44-R54

```
const migrateFakeCollectionRecord = async (record, knex) => {
  const updatedRecord = translateApiCollectionToPostgresCollection(record);
  await knex('collections').insert(updatedRecord);
};

const fakeEncryptFunction = async () => 'fakeEncryptedString';

const migrateFakeProviderRecord = async (record, knex) => {
  const updatedRecord = await translateApiProviderToPostgresProvider(record, fakeEncryptFunction);
  await knex('providers').insert(updatedRecord);
};
```
Sorry if I didn't make that more clear!

Reply:
Awesome, well done!